src/Value.ml

```diff
@@ -39,4 +45,40 @@ let of_string (s : string) : t =
   with Invalid_argument _ -> try
     String String.(chop_suffix_exn ~suffix:"\"" (chop_prefix_exn ~prefix:"\"" s))
   with Invalid_argument _ ->
-    raise (Parse_Exn ("Failed to parse value `" ^ s ^ "`."))
\ No newline at end of file
+    raise (Parse_Exn ("Failed to parse value `" ^ s ^ "`."))
+
+let rec parse_array (acc: (t*t) list) (sexp: Sexp.t) : Type.t * Type.t * (t*t) list *t =
```

Comment:
Can you leave a brief (few lines) comment above this line with the format that you expect Array values to have? Also, may be add a link to our Google doc and point out which of those formats we handle and which ones we don't.

Reply:
Since the document on the format explains what we support and what format we do not. I haave left the following comment ```(* We currently support parsing of array z3 mdoels detailed here https://bit.ly/37DmLSo *) ``` Is this this good enough?

Reply:
Hmm, I looked at the doc, but it doesn't explicitly state that we only handle if-then-else functions. Could you instead add this:

```
(* We assume that an array serialization provides explicit (k,v) pairs --
 * either using nested `store` calls, or if-then-else constructs.
 * The different array formats are described in more details here:
 * https://docs.google.com/document/d/1zSXs91eeJ1hc7bmcUTzeJtjiCHTNEYZKZpi_436HvbA *)
```